packages/backend-core/src/security/roles.ts

```diff
@@ -157,13 +167,16 @@ export async function getRole(
     role = cloneDeep(
       Object.values(BUILTIN_ROLES).find(role => role._id === roleId)
     )
+  } else {
+    // make sure has the prefix (if it has it then it won't be added)
+    roleId = generateRoleID(roleId)
```

Comment:
NAB: The name of the function is a little strange to me. It's called `generateRoleID` yet you pass the roleId into it.

Maybe it should be called something like `prefixRoleID`